pkg/daemons/agent/agent.go

```diff
@@ -185,6 +185,7 @@ func defaultKubeletConfig(cfg *daemonconfig.Agent) (*kubeletconfig.KubeletConfig
 		NodeStatusReportFrequency:        metav1.Duration{Duration: time.Minute * 5},
 		NodeStatusUpdateFrequency:        metav1.Duration{Duration: time.Second * 10},
 		ProtectKernelDefaults:            cfg.ProtectKernelDefaults,
+		ReadOnlyPort:                     0,
```

Comment:
You read the k/k issue. This still won't marshal in the output, why bother?

Reply:
Check the output on disk, and tell me what you see as the value of this field when the config is marshalled to yaml.

Reply:
RIght so why did you add it back? You could probably resolve this by just making your other changes without reverting the change that removed this.